app/models/user.rb

```diff
@@ -9,4 +9,12 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  # Friendship associatons
+
+  has_many :friendships, class_name: 'Friendship', foreign_key: 'confirmer_id'
+  has_many :confirmers, through: :friendships, source: :confirmer
+
+  has_many :friendships, class_name: 'Friendship', foreign_key: 'requester_id'
+  has_many :requesters, through: :friendships, source: :requester
```

Comment:
I can't try this to see if it's working, but I'm almost certain it doesn't.
When you create associations with the same name, the second one overrides the first one. 
You should always create unique names for your associations to avoid that problem.
Imagine rails trying to understand what you mean when you type: `user.friendships` You have two options!

The simplest way is to create two columns on your friendship table: a user_id and a friend_id
Then, you just add a regular association like `has_many :friendships` and the other side `has_many :inverse_friendships, class_name: "Friendship", foreign_key: "friend_id"`